pyo3-macros/src/lib.rs

```diff
@@ -35,10 +35,26 @@ use syn::{parse::Nothing, parse_macro_input, Item};
 /// [1]: https://pyo3.rs/latest/module.html
 #[proc_macro_attribute]
 pub fn pymodule(args: TokenStream, input: TokenStream) -> TokenStream {
-    parse_macro_input!(args as Nothing);
     match parse_macro_input!(input as Item) {
-        Item::Mod(module) => pymodule_module_impl(module),
-        Item::Fn(function) => pymodule_function_impl(function),
+        Item::Mod(module) => {
+            let is_submodule = match parse_macro_input!(args as Option<syn::Ident>) {
+                Some(i) if i == "submodule" => true,
```

Comment:
In most of our argument parsing, we allow the two following forms:

```rust
#[pymodule]
#[pyo3(submodule)]

// or

#[pymodule(submodule)]
```

i.e. I think of the latter as short for the former.

I think it would be nice to do that here, as it would make the UX more consistent with `#[pyclass]` and `#[pyfunction]`.

I'm ok with that also being a follow-up.

Reply:
Let me see if I can easily work that in.

Reply:
Done, that was easy :-)